client/session/profile_manager.go

```diff
@@ -0,0 +1,242 @@
+			return fmt.Errorf("failed to unlink current config file: %v", err)
+		}
+	}
+
+	if err := os.Symlink(m.ProfilePath(profileAlias), m.CurrentPath); err != nil {
+		return fmt.Errorf("could not symlink the config file for %s: %w", profileAlias, err)
+	}
+
+	return nil
+}
+
+// Create adds a new Spacelift profile.
+func (m *ProfileManager) Create(profile *Profile) error {
+	if err := validateProfile(profile); err != nil {
+		return err
+	}
+
+	if err := writeProfileToFile(profile, m); err != nil {
+		return err
+	}
+
+	setCurrentProfile(m.ProfilePath(profile.Alias), m)
+
+	return nil
+}
+
+// Delete removes the profile with the specified alias, and un-selects it as the current profile
+// if it was selected.
+func (m *ProfileManager) Delete(profileAlias string) error {
+	if profileAlias == "" {
+		return errors.New("a profile alias must be specified")
+	}
+
+	if _, err := os.Stat(m.ProfilePath(profileAlias)); err != nil {
+		return fmt.Errorf("no profile named '%s' exists", profileAlias)
+	}
+
+	if err := os.Remove(m.ProfilePath(profileAlias)); err != nil {
+		return err
+	}
+
+	currentTarget, err := os.Readlink(m.CurrentPath)
+
+	switch {
+	case os.IsNotExist(err):
+		return nil
+	case err == nil && currentTarget == m.ProfilePath(profileAlias):
+		return os.Remove(m.CurrentPath)
+	default:
+		return err
+	}
+}
+
+// ProfilePath returns the path to the profile with the specified alias.
+func (m *ProfileManager) ProfilePath(profileAlias string) string {
+	return filepath.Join(m.ProfilesDirectory, profileAlias)
+}
+
+func validateProfile(profile *Profile) error {
+	if profile == nil {
+		return errors.New("profile must not be nil")
+	}
+
+	if profile.Alias == "" {
+		return errors.New("a profile alias must be specified")
+	}
+
+	for _, invalidAlias := range invalidProfileAliases {
+		if strings.Contains(profile.Alias, invalidAlias) {
+			return fmt.Errorf("'%s' is not a valid profile alias", profile.Alias)
+		}
+	}
+
+	switch credentialType := profile.Credentials.Type; credentialType {
+	case CredentialsTypeGitHubToken:
+		if err := validateGitHubCredentials(profile); err != nil {
+			return err
+		}
+
+	case CredentialsTypeAPIKey:
+		if err := validateAPIKeyCredentials(profile); err != nil {
+			return err
+		}
+
+	default:
+		return fmt.Errorf("'%d' is an invalid credential type", credentialType)
+	}
+
+	return nil
+}
+
+func validateGitHubCredentials(profile *Profile) error {
+	if profile.Credentials.AccessToken == "" {
+		return errors.New("'AccessToken' must be provided for GitHub token credentials")
+	}
+
+	return nil
+}
+
+func validateAPIKeyCredentials(profile *Profile) error {
+	if profile.Credentials.KeyID == "" {
+		return errors.New("'KeyID' must be provided for API Key credentials")
+	}
+
+	if profile.Credentials.KeySecret == "" {
+		return errors.New("'KeySecret' must be provided for API Key credentials")
+	}
+
+	return nil
+}
+
+func setCurrentProfile(profilePath string, manager *ProfileManager) error {
```

Comment:
passing `*ProfileManager` is counter intuitive.
It would make more sense to make this a method on the `ProfileManager`.
```
func (pm *ProfileManageer) setCurrentProfile
```
Same for `func writeProfileToFile(profile *Profile, manager *ProfileManager)`

Reply:
Cheers - I honestly didn't even consider that!